internal/common/handler.go

```diff
@@ -9,6 +9,8 @@ import (
 // `opaque` can be set to some framework-specific struct - for example, gin.Context.
 //
 // Prefer statuses 200 (success), 400 (generic, expected error), and 500 (generic, unexpected error).
+//
+// If a HandlerFn returns empty data, drivers should not write any data to the response.
```

Comment:
I thought there was at least one case where we did write data when the handler returned nothing, but I'm not seeing it. Maybe that was for the old-style endpoints.

Anyway, does it hurt anything if we _do_ write something when the handler wrote nothing?

Reply:
> does it hurt anything if we do write something when the handler wrote nothing?

No, but I'm not sure that's the question you intended to ask. If a driver writes data when the handler wrote nothing, everything's normal. This is the happy path for generic handlers.

If the driver writes data when the handler _returns_ empty data, then the handler may have already written something (like in the case of `http.Redirect` for unvalidated-redirect). If the handler has written data and the driver attempts to call `WriteHeader`, an error will be logged from: https://github.com/golang/go/blob/180bcad33dcd3d59443fe8eda5ae7556b1b2945b/src/net/http/server.go#L1147

I can tighten the doc comment to say "drivers should not call WriteHeader on the response" but that may cause more confusion. I'm leaning toward leaving it as-is for now. Let me know if you have suggestions.

Reply:
> does it hurt anything if we do write something when the handler wrote nothing?

No, but I'm not sure that's the question you intended to ask. If a driver writes data when the handler wrote nothing, everything's normal. This is the happy path for generic handlers.

If the driver writes data when the handler _returns_ empty data, then the handler may have already written something (like in the case of `http.Redirect` for unvalidated-redirect). If the handler has written data and the driver attempts to call `WriteHeader`, an error will be logged from: https://github.com/golang/go/blob/180bcad33dcd3d59443fe8eda5ae7556b1b2945b/src/net/http/server.go#L1147

I can tighten the doc comment to say "drivers should not call WriteHeader on the response" but that may cause more confusion. I'm leaning toward leaving it as-is for now. Let me know if you have suggestions.

Reply:
My bad, yes I was thinking `returns` when I wrote `wrote`.